### PR TITLE
tweaks to the monerov1 cryptonight algorithm

### PR DESCRIPTION
**Pull request by vtnerd**

### CODE REVIEW

File: src/crypto/slow-hash.c [hunk 1]
```diff
@@ -61,16 +61,33 @@ extern int aesb_pseudo_round(const uint8_t *in, uint8_t *out, const uint8_t *exp
...
+
+#define NONCE_POINTER (((const uint8_t*)data)+35)
+
+#define VARIANT1_PORTABLE_INIT() \
+  uint8_t tweak1_2[8]; \
+  do if (variant > 0) \
+  { \
+    VARIANT1_CHECK(); \
+    memcpy(&tweak1_2, &state.hs.b[192], sizeof(tweak1_2)); \
+    xor64(tweak1_2, NONCE_POINTER); \
```
Comment (psychocrypt) [inline src/crypto/slow-hash.c:82]: Why do you XOR with the address of the pointer? Should it be `*((const uint64_t*)NONCE_POINTER))`

Comment (SChernykh) [reply src/crypto/slow-hash.c:82]: @psychocrypt VARIANT1_PORTABLE_INIT uses xor64 from line 1267 - it gets two uint8_t pointers

**Comment by psychocrypt** [comment]
Thx I missed that there are three different versions of xor64 in the diff.

**Comment by moneromooo-monero** [comment]
Merged in the Monero PR.

**Pull request closed without merging**